### PR TITLE
Bump to NVML version 11.8

### PR DESCRIPTION
**Pull request by klueska**

### CODE REVIEW

File: gen/nvml/nvml.h [hunk 1]
```diff
@@ -627,9 +627,9 @@ typedef struct
     struct
     {
         nvmlThermalController_t controller;
-        unsigned int defaultMinTemp;
-        unsigned int defaultMaxTemp;
-        unsigned int currentTemp;
+        int defaultMinTemp;
```
Comment (elezar) [inline gen/nvml/nvml.h:630]: This is internal only from our perspective, correct?

**Review by elezar — COMMENTED** [review]
Looks good, but let's wait for a rebase on #52 since I only looked at the last three commits.

File: gen/nvml/nvml.h (hunk 1 above)
Comment (klueska) [reply gen/nvml/nvml.h:630]: What do you mean by internal only? The file here in the `gen` directory should be a direct copy from what's upstream.

File: gen/nvml/gpm.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nvml
+
+// nvml.GpmMetricsGet()
```
Comment (elezar) [inline gen/nvml/gpm.go:17]: Is this comment at the right place?

File: gen/nvml/gpm.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+type GpmMetricsGetVType struct {
```
Comment (elezar) [inline gen/nvml/gpm.go:18]: Where is this type actually used? Similarly for `GpmSupport*` below.

File: gen/nvml/nvml.h (hunk 1 above)
Comment (elezar) [reply gen/nvml/nvml.h:630]: I mean that the change of type doesn't require other changes in our wrapper.

File: gen/nvml/gpm.go (hunk 3 above)
Comment (klueska) [reply gen/nvml/gpm.go:18]: Yeah. I wanted it at the top of the full set of functions related to `GpmMetricsGet()`. I could move the actual `GpmMetricsGet()` to the top though if you think that's clearer.

File: gen/nvml/gpm.go (hunk 2 above)
Comment (klueska) [reply gen/nvml/gpm.go:17]: Yeah. I wanted it at the top of the full set of functions related to GpmMetricsGet(). I could move the actual GpmMetricsGet() to the top though if you think that's clearer.

Comment (elezar) [reply gen/nvml/gpm.go:17]: I would think it's a little clearer in terms of docstring-like usage. It's not a blocker though.

File: gen/nvml/gpm.go (hunk 3 above)
Comment (klueska) [reply gen/nvml/gpm.go:18]: It's the return value of a call to `GpmMetricsGetV()`.

The usage is:
```
var MetricsGet GpmMetricsGetType)
ret := GpmMetricsGetV(&MetricsGet).V1()
```

Comment (elezar) [reply gen/nvml/gpm.go:18]: I suppose my question is where do we expliclity need it, or is this something that a user needs to take care of. If that is the case we may want to add additional documentation to the affected functions.

Comment (klueska) [reply gen/nvml/gpm.go:18]: It is the pattern I have been following for these new "versioned" structs that the NVML team has been introducing. The API using the latest struct version is always available using e.g. `GpmMetricsGet(...)`, but if you want to explicitly use a different version you can call e.g. `GpmMetricsGetV(...).V1()`, `GpmMetricsGetV(...).V2(), ... instead.

I first started doing it with NVMl 11.6 here:
https://github.com/NVIDIA/go-nvml/pull/40/files#diff-0298839f64113aa3439ac75706f86ef591ba5c852ef6a740e30fb52322471488R1974

File: gen/nvml/nvml.h (hunk 1 above)
Comment (klueska) [reply gen/nvml/nvml.h:630]: It shouldn't since they are struct fields set by the end user, not us.

File: gen/nvml/gpm.go (hunk 3 above)
Comment (elezar) [reply gen/nvml/gpm.go:18]: OK. So a user *could* use these once we have multiple versions but it is not required here.

**Review by elezar — APPROVED** [review]
I think my comments on docstrings are minor and should not block this.

**Pull request merged**